modules/options.nix

```diff
@@ -13,15 +13,35 @@ with lib; {
         VISUAL = "nvim";
       };
       description = lib.mdDoc ''
-        A set of environment variables. The value of each variable can be either a string or a list of
-        strings.  The latter is concatenated, interspersed with colon
-        characters.
+        A set of environment variables. The value of each variable can be either
+        a string or a list of strings.  The latter is concatenated, interspersed
+        with colon characters.
+
+        Alternatively, this may be a list of sets of environment variables. In
+        that case, order of the variables is preserved, and values are not
+        escaped, which means variables may be evaluated at runtime.
       '';
-      type = with types; attrsOf (either str (listOf str));
-      apply = mapAttrs (n: v:
-        if isList v
-        then concatStringsSep ":" v
-        else v);
+      type = with types; let
+        envAttrSet =
+          # {
+          #   FOO = "BAR";
+          #   PATH = ["/bin" "/usr/bin"];
+          # }
+          attrsOf (either str (listOf str));
+      in
+        either envAttrSet (listOf envAttrSet);
```

Comment:
I don't love how this will prevent merging of modules, but I think time has been called on discussion and that's just the way it's going to be for now